storage/gc_queue.go

```diff
@@ -239,10 +232,23 @@ func (gcq *gcQueue) process(now proto.Timestamp, repl *Replica) error {
 		}
 	}
 
+	done := true
 	if len(intents) > 0 {
+		done = false
 		repl.resolveIntents(repl.context(), intents)
 	}
 
+	// Set start and end keys.
+	if len(gcArgs.Keys) > 0 {
+		done = false
+		gcArgs.Key = gcArgs.Keys[0].Key
+		gcArgs.EndKey = gcArgs.Keys[len(gcArgs.Keys)-1].Key.Next()
+	}
+
+	if done {
+		return nil
+	}
+
 	// Send GC request through range.
```

Comment:
We can only send the GC request if `gcArgs.Keys` is not empty. I think the old logic was right to return when `gcArgs.Keys` was empty; the only thing we need to change is move `repl.resolveIntents` up before that return.